test/ask/runtime/survey_broker_test.exs

```diff
@@ -1885,15 +1868,23 @@ defmodule Ask.Runtime.SurveyBrokerTest do
     refute respondent.state == state
   end
 
-  def wait_all_cancellations_from_pids(pids) do
-    pids
-    |> Enum.map(fn {_, pid} -> Process.monitor(pid) end)
-    |> Enum.each(&receive_down/1)
-  end
+  def wait_all_cancellations(%{id: survey_id}) do
+    ref =
+      Ask.Runtime.SurveyCancellerSupervisor.canceller_pid(survey_id)
+      |> Process.monitor()
 
-  def receive_down(ref) do
     receive do
       {:DOWN, ^ref, _, _, _} -> :task_is_down
     end
   end
+
+  defp assert_disposition_changed(respondent_id, old_disposition, new_disposition) do
+    last_entry =
+      Repo.one(from log in SurveyLogEntry, where: log.respondent_id == ^respondent_id, where: log.action_type == "disposition changed", order_by: [desc: :id], limit: 1)
+
+    assert last_entry.disposition == to_string(old_disposition)
+    assert last_entry.action_data == upcaseFirst(new_disposition)
+  end
+  defp upcaseFirst(value) when is_atom(value), do: to_string(value) |> upcaseFirst
+  defp upcaseFirst(<<first::utf8, rest::binary>>), do: String.upcase(<<first::utf8>>) <> rest
```

Comment:
💅 Perhaps instead of duplicating these functions here from `survey_test` we can move it to `test/support/test_helpers.ex`